backend/database/versions/05_67c4015bce73_add_datasets_custom_ordered_to_.py

```diff
@@ -0,0 +1,27 @@
+"""add datasets custom ordered to collection version
+
+Revision ID: 05_67c4015bce73
+Revises: 04_2f30f3bcc9aa
+Create Date: 2024-01-25 11:42:46.126701
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "05_67c4015bce73"
+down_revision = "04_2f30f3bcc9aa"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column(
+        "CollectionVersion",
+        sa.Column("datasets_custom_ordered", sa.Boolean(), nullable=False, server_default=sa.sql.expression.false()),
```

Comment:
We're using `sa.BOOLEAN()` in all other places (a total of 2) in the migrations where we need a default bool. It's slightly different based on documentation. It's not totally clear to me which choice is best; I'd have to read more. But let's change to `sa.BOOLEAN()` for consistency (also: we know it works already 😅 ).

Reply:
Actually, I'm noticing that in the other two instances we allow `nullable=True` and we do not set a `server_default` value. I'm thinking we should do this.

Reply:
Thanks for noticing `Boolean` vs `BOOLEAN`! I had a look at the docs: `BOOLEAN` uses the [SQL type](https://docs.sqlalchemy.org/en/20/core/type_basics.html#sql-standard-and-multiple-vendor-uppercase-types); `Boolean` uses a [generic type](https://docs.sqlalchemy.org/en/20/core/type_basics.html#generic-camelcase-types). 

> Actually, I'm noticing that in the other two instances we allow nullable=True and we do not set a server_default value. I'm thinking we should do this.

Regarding making `datasets_custom_ordered` `nullable=True`, wouldn't we want the database to protect against NULL values being inserted? Happy to move it to `nullable=True` but want to double-check this is the desired approach.

Reply:
I say this because at present the other boolean values are configured this way (even though we otherwise enforce a True or False value for all rows). I know there is some tangentially-related issue with the way we treat boolean values when we extract rows from psql (we have some code somewhere that tests for `is not False` or something like that, rather than `is True`...)

cc @nayib-jose-gloria @Bento007 for comment on these patterns before I get a chance to look more into it